ocaml/xapi/xapi_pool.ml

```diff
@@ -665,9 +665,17 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
            ( Api_errors.license_restriction
            , [Features.name_of_feature Features.Pool_size] ))
   in
+  let assert_pool_master_not_in_updating () =
+    let master = get_master ~rpc ~session_id in
```

Comment:
Perhaps just block if `apply_updates` is in `pool.current_operations`. It's best not to add new hosts to the pool if anything is updating.